### PR TITLE
DRILL-8114: Prevent applying Iceberg project on non-Iceberg tables

### PR DESCRIPTION
**Pull request by vvysotskyi**
# [DRILL-8114](https://issues.apache.org/jira/browse/DRILL-8114): Prevent applying Iceberg project on non-Iceberg tables

## Description
Added checks to ensure that table has iceberg group scan when matching iceberg rules (the same for mongo).

## Documentation
NA

## Testing
Added unit tests.

### CODE REVIEW

File: contrib/format-iceberg/src/main/java/org/apache/drill/exec/store/iceberg/plan/IcebergPluginImplementor.java [hunk 1]
```diff
@@ -144,14 +141,18 @@ public boolean splitProject(Project project) {
...
-    return true;
+    return hasIcebergGroupScan(project);
   }
 
   @Override
   public GroupScan getPhysicalOperator() {
     return groupScan;
   }
 
+  private boolean hasIcebergGroupScan(RelNode node) {
```
Comment (luocooong) [inline contrib/format-iceberg/src/main/java/org/apache/drill/exec/store/iceberg/plan/IcebergPluginImplementor.java:152]: It seems that this validation rule also applies to #2389?

Comment (vvysotskyi) [reply contrib/format-iceberg/src/main/java/org/apache/drill/exec/store/iceberg/plan/IcebergPluginImplementor.java:152]: Yes, this PR also fixes the issue from that PR.

File: contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/plan/MongoPluginImplementor.java [hunk 2]
```diff
@@ -292,7 +297,11 @@ public GroupScan getPhysicalOperator() {
       newSpec, columns, runAggregate);
   }
 
-  public static int direction(RelFieldCollation fieldCollation) {
+  private boolean hasMongoGroupScan(RelNode node) {
```
Comment (luocooong) [inline contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/plan/MongoPluginImplementor.java:300]: @vvysotskyi I have an idea, could we transform that function as a base or interface function (such the `hasXXXGroupscan()`)? Ensure the new format based on this framework is needed to implement this filter function.

Comment (vvysotskyi) [reply contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/plan/MongoPluginImplementor.java]: Thanks, done.

**Review by luocooong — APPROVED** [review]
@vvysotskyi Thanks for the fix, your efforts make Drill more stable.

PS: Close both DRILL-8060 and DRILL-8058 once we close this JIRA ticket.

**Comment by vvysotskyi** [comment]
@luocooong, thanks for the review!

**Pull request merged**